p2p/src/net/mock/backend.rs

```diff
@@ -444,12 +452,53 @@ where
                 version,
                 subscriptions,
                 receiver_address,
+                handshake_nonce,
             } => {
-                let PendingPeerContext { address, role, tx } =
-                    self.pending.remove(&peer_id).expect("peer to exist");
+                let PendingPeerContext {
+                    address,
+                    peer_role,
+                    tx,
+                } = match self.pending.remove(&peer_id) {
+                    Some(pending) => pending,
+                    // Might be removed if self-connection detected
+                    None => return Ok(()),
+                };
+
+                if peer_role == PeerRole::Inbound {
+                    // Look for own outbound connection with same nonce
+                    let outbound_peer_id = self
+                        .pending
+                        .iter()
+                        .find(|(_peer_id, pending)| {
+                            pending.peer_role == PeerRole::Outbound { handshake_nonce }
+                        })
+                        .map(|(peer_id, _pending)| *peer_id);
+
+                    if let Some(outbound_peer_id) = outbound_peer_id {
+                        let outbound_pending =
+                            self.pending.remove(&outbound_peer_id).expect("peer must exists");
+
+                        log::info!(
+                            "self-connection detect on address {:?}",
+                            outbound_pending.address
+                        );
+
+                        // Report outbound connection failure
+                        self.conn_tx
+                            .send(ConnectivityEvent::ConnectionError {
+                                address: outbound_pending.address,
+                                error: P2pError::DialError(DialError::AttemptToDialSelf),
+                            })
+                            .await
+                            .map_err(P2pError::from)?;
+
+                        // Nothing else to do, just drop inbound connection
+                        return Ok(());
+                    }
+                }
 
-                match role {
-                    Role::Outbound => {
+                match peer_role {
+                    PeerRole::Outbound { .. } => {
```

Comment:
Please don't use `..`. If you don't need the variable, put explicitly a `_`. The problem with `..` is that it prevents the compiler from reacting when you add more variables in the future.

Reply:
Sure, fixed.